src/main/java/GooglePhotosAutoUploader.java

```diff
@@ -0,0 +1,31 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class GooglePhotosAutoUploader {
+    private Object LOCK = new Object();
+    private List<String> photosToUploadList = new ArrayList<>();
+    public void startAutoUpload() {
+        synchronized (LOCK) {
+            if (photosToUploadList.isEmpty()) {
+                try {
+                    LOCK.wait();
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e.getMessage());
+                }
+            }
+            uploadPhotos();
+        }
+    }
+    public void onNewPhotoAdded(String photoPath) {
+        synchronized (LOCK) {
+            photosToUploadList.add(photoPath);
+            LOCK.notify();
+            Thread.sleep(1000);
+        }
+
+        public void uploadPhotos() {
+            System.out.println("Начинаем загрузку...");
+            photosToUploadList.forEach(photoToUpload -> System.out.println("Фото " + photoToUpload + " загружается..."));
+            System.out.println("Загрузка завершена!");
+        }
```

Comment:
Согласно постановке, необходимо очищать список для загрузки, в текущем варианте будем грузить одни и те же файлы много раз, программа не будет работать корректно

Reply:
Я бы предложил использовать следующий подход:
```suggestion
        private void uploadPhotos() {
        System.out.println("Начинаем загрузку...");
        Iterator<String> iterator = photosToUpload.iterator();
        while (iterator.hasNext()) {
            String photo = iterator.next();
            System.out.println("Фото " + photo + " загружается...");
            System.out.println("Фото " + photo + " загружено...");
            iterator.remove();
        }
        System.out.println("Загрузка завершена!");
    }
```

При таком сценарии мы сразу удаляемый загруженный файл, чтобы избежать неконсистентного состояния (поток может умереть по какой-либо причине, в итоге в списке останутся файлы, которые уже загрузили ранее)